driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java

```diff
@@ -188,29 +194,26 @@ private boolean eventRequired() {
         return commandListener != null;
     }
 
-    private void appendCommonLogFragment(final List<Entry> entries, final StringBuilder builder) {
-        builder.append(" using a connection with driver-generated ID %d");
-        entries.add(new Entry("driverConnectionId", description.getConnectionId().getLocalValue()));
-
-        Integer connectionServerValue = description.getConnectionId().getServerValue();
-        if (connectionServerValue != null) {
-            builder.append(" and server-generated ID %d");
-            entries.add(new Entry("serverConnectionId", connectionServerValue));
-        }
-
-        builder.append(" to %s:%s");
-        entries.add(new Entry("serverHost", description.getServerAddress().getHost()));
-        entries.add(new Entry("serverPort", description.getServerAddress().getPort()));
-
-        ObjectId descriptionServiceId = description.getServiceId();
-        if (descriptionServiceId != null) {
-            builder.append(" with service ID %s");
-            entries.add(new Entry("serviceId", descriptionServiceId));
-        }
+    private void logEventMessage(final String messagePrefix, final String messageId, @Nullable final Throwable exception,
+                                 final Consumer<Deque<Entry>> entriesMutator) {
+        String format = messagePrefix + " using a connection with driver-generated ID {}"
+                + "[ and server-generated ID {}] to {}:{}[ with service ID {}]. The request ID is {}"
+                + " and the operation ID is {}.[ Command: {}]";
+        logEventMessage(messageId, exception, entriesMutator, format);
+    }
 
-        builder.append(". The request ID is %s and the operation ID is %s.");
-        entries.add(new Entry("requestId", message.getId()));
-        entries.add(new Entry("operationId", operationContext.getId()));
+    private void logEventMessage(final String messageId, final @Nullable Throwable exception, final Consumer<Deque<Entry>> entriesMutator,
```

Comment:
The calling code would be less confusing if it could pass two mutators: a prefix mutator and a suffix mutator, where the first is called before the common entries are added and the second called after they are added. That would avoid the need to call `addFirst` twice in the opposite order that you would expect.  Both mutators could just call `Deque#add`.